application/forms/EntryForm.php

```diff
@@ -311,114 +317,233 @@ protected function assemble()
     public function addEntry(int $scheduleId): void
     {
         $data = $this->formValuesToDb();
+        $membershipHash = $this->getMembershipHashFromRecipients($scheduleId);
         $recipients = array_map(function ($recipient) {
             return explode(':', $recipient, 2);
         }, explode(',', $this->getValue('recipient')));
 
         $db = Database::get();
         $db->beginTransaction();
 
-        $db->insert('timeperiod', ['owned_by_schedule_id' => $scheduleId]);
-        $timeperiodId = $db->lastInsertId();
+        $members = ScheduleMember::on(Database::get())
+            ->with('timeperiod')
+            ->filter(
+                Filter::all(
+                    Filter::equal('timeperiod.owned_by_schedule_id', $scheduleId),
+                    Filter::equal('schedule_member.membership_hash', $membershipHash)
+                )
+            );
+
+        if ($members->count() > 0) {
```

Comment:
I'd rather select from `Timeperiod` directly. That's the only information you're really interested in. The count of members for example, you're using it to check whether there are any members at all. That's a common mistake, I mean, to count although the count itself is of no interest. So by fetching the timeperiod directly, you already verified that at least one member's hash matches. (Given that the filter is still the same)